solr/core/src/test/org/apache/solr/handler/admin/api/RequestCoreCommandStatusTest.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.handler.admin.api;
+
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Map;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.Response;
+import org.apache.solr.SolrTestCaseJ4;
+import org.apache.solr.core.CoreContainer;
+import org.apache.solr.handler.admin.CoreAdminHandler;
+import org.apache.solr.jersey.InjectionFactories;
+import org.apache.solr.jersey.SolrJacksonMapper;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.response.SolrQueryResponse;
+import org.glassfish.hk2.api.Factory;
+import org.glassfish.hk2.utilities.binding.AbstractBinder;
+import org.glassfish.jersey.process.internal.RequestScoped;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.test.JerseyTest;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/** Test for {@link RequestCoreCommandStatus}. */
+public class RequestCoreCommandStatusTest extends JerseyTest {
```

Comment:
[-0] I can't recall right now, but in a prior PR I ran into an issue with how Solr's test-framework ran JerseyTest-based test classes.

Let me dig into this a bit more to try to remember what the issue was, but we _might_ have to restructure this to be more of a pure unit test like you wrote in #2128

Reply:
I saw the build passing with this test. I am hoping we will be able to add this to main.
Because I like JerseyTest as they test the real end-points rather calling the method. Wdyt?

Reply:
Agreed - I also really like the "level" that `JerseyTest` allows us to validate - being able to test that the HTTP path, request body, etc. match what you expect is really valuable.  And I've pulled the code down locally and agree that the test passes as-is.  It's a good test 👍 

The problem is the base-class, `JerseyTest`.

Generally in Solr's test suite, we want to support users running any number of tests in parallel (or even running the same test 'N' times in parallel) without any conflicts.  Solr's base classes (SolrTestCase, SolrTestCaseJ4, etc.) have a good bit of logic built into them to support this: locking around how ports are allocated when starting test servers, ensuring system properties are cleared in between tests to avoid any spillover, tracking "Closable" resources to detect memory leaks, etc. 

`JerseyTest` doesn't provide a lot of these "good citizen" guarantees though.  I accidentally broke the build earlier this year when adding a `JerseyTest` subclass for another API.  (I'm hazy on the specifics, but I think that whenever two `JerseyTest`-based tests ran at the same time they ended up "choosing" the same port for the test server and failing.)

It's definitely fixable - we could probably look at what `JerseyTest` is actually doing and come up with a `public class SolrJerseyTest extends SolrTestCase` that brings the best of both worlds together.  But that doesn't exist yet, so I've been trying to use JerseyTest sparingly for now.